src/simsopt/geo/curvexyzhelical.py

```diff
@@ -0,0 +1,119 @@
+    gamma = gamma.at[:, 0].add(x)
+    gamma = gamma.at[:, 1].add(y)
+    gamma = gamma.at[:, 2].add(z)
+    return gamma
+
+
+class CurveXYZHelical(JaxCurve):
+    r'''A curve representation for a helical coil that does not lie on a torus.  The coordinates of the curve are given by:
+        .. math::
+            \hat x(\theta) &= x_{c, 0} + \sum_{m=1}^{\text{order}} x_{c,m} cos(2pi n_{\text{fp}} m \theta)\\
+            \hat y(\theta) &=            \sum_{m=1}^{\text{order}} y_{s,m} sin(2pi n_{\text{fp}} m \theta)\\
+            x(\theta) &= \hat x(\theta)  \cos(2pi \theta) - \hat y(\theta)  \sin(2pi \theta)\\
+            y(\theta) &= \hat x(\theta)  \sin(2pi \theta) + \hat y(\theta)  \cos(2pi \theta)\\
+            z(\theta) &= \sum_{i=1}^{\text{order}} z_{s,m} sin(2pi n_{\text{fp}} \theta)
+        if the coil is stellarator symmetric.  When the coil is not stellarator symmetric, the formulas above
+        become
+        .. math::
+            \hat x(\theta) &= x_{c, 0} + \sum_{m=1}^{\text{order}} x_{c, m} cos(2pi m \theta) +  x_{s, m} sin(2pi m \theta)\\
+            \hat y(\theta) &= y_{c, 0} + \sum_{m=1}^{\text{order}} y_{c, m} cos(2pi m \theta) +  y_{s, m} sin(2pi m \theta)\\
+            x(\theta) &= \hat x(\theta)  \cos(2pi \theta) - \hat y(\theta)  \sin(2pi \theta)\\
+            y(\theta) &= \hat x(\theta)  \sin(2pi \theta) + \hat y(\theta)  \cos(2pi \theta)\\
+            z(\theta) &= z_{c, 0} + \sum_{m=1}^{\text{order}} z_{c, m} cos(2pi \theta) + z_{s, m} sin(2pi \theta)
+        
+        Args:
+            quadpoints: number of grid points/resolution along the curve,
+            order:  how many Fourier harmonics to include in the Fourier representation,
+            nfp: discrete rotational symmetry number, 
+            stellsym: stellaratory symmetry if True, not stellarator symmetric otherwise.
+        '''
+
+    def __init__(self, quadpoints, order, nfp, stellsym, **kwargs):
+        if isinstance(quadpoints, int):
+            quadpoints = np.linspace(0, 1, quadpoints, endpoint=False)
+        pure = lambda dofs, points: jaxXYZHelicalFouriercurve_pure(
+            dofs, points, order, nfp, stellsym)
+        
+        self.order = order
+        self.nfp = nfp
+        self.stellsym = stellsym
+        self.coefficients = np.zeros(self.num_dofs())
+        if "dofs" not in kwargs:
+            if "x0" not in kwargs:
+                kwargs["x0"] = self.coefficients
+            else:
+                self.set_dofs_impl(kwargs["x0"])
+
+        super().__init__(quadpoints, pure, names=self._make_names(order), **kwargs)
+
+    def _make_names(self, order):
+        if self.stellsym:
+            x_cos_names = [f'xc({i})' for i in range(0, order + 1)]
+            x_names = x_cos_names
+            y_sin_names = [f'ys({i})' for i in range(1, order + 1)]
+            y_names = y_sin_names
+            z_sin_names = [f'zs({i})' for i in range(1, order + 1)]
+            z_names = z_sin_names
```

Comment:
```suggestion
            x_names = [f'xc({i})' for i in range(1, order + 1)] # x cos names
            y_names = [f'ys({i})' for i in range(1, order + 1)] # y sin names
            z_names = [f'zs({i})' for i in range(1, order + 1)] # z sin names
```
moreover the range for x_names is starting from 0 instead of 1. Is it correct?